validator/validator.go

```diff
@@ -23,10 +23,13 @@ func validateResourceBounds(fl validator.FieldLevel) bool {
 	}
 
 	version := req.Version.String()
-	if (version == "0x3" || version == "0x100000000000000000000000000000003") &&
-		(req.ResourceBounds == nil || len(*req.ResourceBounds) != 3) {
-		return false
+
+	// Only enforce resource bounds validation for version 0x3
+	if version == "0x3" || version == "0x100000000000000000000000000000003" {
+		return req.ResourceBounds != nil && len(*req.ResourceBounds) == 3
 	}
+
+	// Allow earlier versions without resource bounds
 	return true
 }
```

Comment:
Nitpick: Return the condition instantly:

```suggestion
// Only enforce resource bounds validation for version 0x3
return version != "0x3" && version != "0x100000000000000000000000000000003" || 
       (req.ResourceBounds != nil && len(*req.ResourceBounds) == 3)
```